cli/src/cluster_query.rs

```diff
@@ -1827,6 +1843,71 @@ pub fn process_transaction_history(
     Ok(transactions_found)
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct CliRentCalculation {
+    pub lamports_per_byte_year: u64,
+    pub lamports_per_epoch: u64,
+    pub rent_exempt_minimum_lamports: u64,
+}
+
+impl CliRentCalculation {
+    fn _write_str(
+        w: &mut dyn fmt::Write,
+        per_byte_year: &str,
+        per_epoch: &str,
+        exempt_minimum: &str,
+    ) -> fmt::Result {
+        writeln_name_value(w, "Rent per byte-year:", per_byte_year)?;
+        writeln_name_value(w, "Rent per epoch:", per_epoch)?;
+        writeln_name_value(w, "Rent-exempt minimum:", exempt_minimum)
+    }
+}
+
+impl fmt::Display for CliRentCalculation {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let per_byte_year = format!("{}", Sol(self.lamports_per_byte_year));
+        let per_epoch = format!("{}", Sol(self.lamports_per_epoch));
+        let exempt_minimum = format!("{}", Sol(self.rent_exempt_minimum_lamports));
+        Self::_write_str(f, &per_byte_year, &per_epoch, &exempt_minimum)
+    }
+}
+
+impl QuietDisplay for CliRentCalculation {}
+impl VerboseDisplay for CliRentCalculation {
+    fn write_str(&self, w: &mut dyn std::fmt::Write) -> std::fmt::Result {
+        let per_byte_year = format!("{} Lamports", self.lamports_per_byte_year);
+        let per_epoch = format!("{} Lamports", self.lamports_per_epoch);
+        let exempt_minimum = format!("{} Lamports", self.rent_exempt_minimum_lamports);
```

Comment:
`build_balance_message()` here too, please.

Reply:
Actually, strike that. I'm not sure that it's intuitive to request `verbose` to get output in lamports.
I'd prefer to add the `--lamports` arg so people can request lamports specifically. (In which case, you'll only need one Display implementation, and can get rid of `_write_str()`)